api/acl.go

```diff
@@ -1047,7 +1046,7 @@ func (a *ACL) RulesTranslate(rules io.Reader) (string, error) {
 	parseQueryMeta(resp, qm)
 	qm.RequestTime = rtt
 
-	ruleBytes, err := ioutil.ReadAll(resp.Body)
+	ruleBytes, err := io.ReadAll(resp.Body)
```

Comment:
Usually we handle go version compatibility differently in the `api` package so that folks using the library aren't forced to upgrade to a future version before their ready. I think the policy seems to be assuming folks are using a go version that  is still receiving security updates (which 1.16 is definitely NOT anymore).

So in this case it's fine to do the swap. I wonder if we should correlate this with bumping the `go` directive in `api/go.mod` from `1.12` to at least `1.16` or maybe all the way to `1.18` (which is the oldest version we test against in CI).

Reply:
The same logic would apply to `sdk/go.mod` too.

Reply:
👍🏽 Can do, I think `1.18` makes sense here then.
There is a *ton* of module tidying, save that for another PR?